File: php/src/Advertisements/User/Domain/Exceptions/MemberAlreadyExistsException.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisements\User\Domain\Exceptions;

use Demo\App\Common\Domain\DomainException;

final class MemberAlreadyExistsException extends DomainException
{
    private const string MEMBER_EXISTS_MESSAGE = 'Member already exists';

    private function __construct(string $message)
    {
        parent::__construct($message);
    }

    public static function build(): self
    {
        return new self(self::MEMBER_EXISTS_MESSAGE);
    }

    public function message(): string
    {
        return $this->getMessage();
    }
}

File: kotlin/src/main/kotlin/advertisements/user/domain/exceptions/MemberAlreadyExistsException.kt
package advertisements.user.domain.exceptions

import common.domain.DomainException


class MemberAlreadyExistsException private constructor(message: String) : DomainException(message) {

    companion object {
        private const val MEMBER_EXISTS_MESSAGE = "Member already exists"

        fun build(): MemberAlreadyExistsException {
            return MemberAlreadyExistsException(MEMBER_EXISTS_MESSAGE)
        }
    }

    fun message(): String {
        return this.message ?: ""
    }
}